### ISSUE TITLE
[FR] frametime graph scaling - current graph is often very misleading

### ISSUE THREAD

**Issue by pallaswept** [flightlessmango/MangoHud#1397]
This is an FR because fixing this requires new features, but there is a real problem here. Some might call it a bug, I guess it's intentional, so is a feature.

The frametime plot we have now can be very misleading. It generally shows an excessively large range for the Y axis, meaning that important frame timing problems are masked, for example smaller (but still important especially at high frame rates) variations can be smaller than 1 pixel in height and cannot be seen. The result is a plot which can look like it's a pretty flat line, but actually, isn't.

Here's an example from [a random video I just watched](https://www.youtube.com/watch?v=O_31WWSzf4I): 
![image](https://github.com/user-attachments/assets/fc5b3a29-6c6b-4268-8963-a93a8313f744)

Note how the line appears rather flat. But look at the frametime variation. minimum 7ms maximum 12.6, It's absolutely massive, the frametime variation is almost an entire frame. This is suffering serious performance problems, this plot ought to look like a mountain range, or at least a flat line with some big spikes in it (who knows, we can't see from this plot)... but it doesn't, everything appears as if it's OK. This is EXTREMELY common, in fact, I see this in almost every video I can find of games with mangohud running - the graph says "smooth", the numbers say the graph lies.

An obvious solution to this is auto-scaling the plot to the height of the largest sample, and this has been proven to work well in other similar applications (eg RTSS on Windows has this option). Alternatively to some kind of auto-scaling, providing a fixed scale to the user would be necessary, to make this plot more functional - although this becomes very limited when using different games. Perhaps a developer might have a clever idea for something new, like, a plot which scales itself to appropriate sizes depending on average FPS or something. 

I guess, I don't want to request a certain solution, because perhaps someone has a better idea... but there's definitely a problem to be fixed here, so I thought I should mention it. To me, auto-scaling as per RTSS is the way forward, but I'll leave the specifics to someone more educated on this tool. Thanks in advance for any solution to this.

**Comment by flightlessmango** [comment]
>An obvious solution to this is auto-scaling the plot to the height of the largest sample

This option already exists `dynamic_frame_timing`

**Comment by pallaswept** [comment]
Hah, I almost edited that post to say:

Edit: Just in case I'm an idiot and this already exists, my FR is that it should be on by default, to avoid misleading results such as the above.

Glad to be an idiot today :D Thanks @flightlessmango

**Comment by pallaswept** [comment]
> my FR is that it should be on by default, to avoid misleading results such as the above.

In addition, it would be great to add this feature and any others that might be missing, to the example config! 
Maybe it might be good to edit the readme a little, too.

I'll share my experience, to explain:
I read the readme, and when I got to the configuration section, the first thing I read is that the example config file can be used to set up MangoHud.
Then, there is a section about environment variables. All the config file entries are listed under this heading. 
For this reason, I was confused and did not even realise that this was a list of config file entries.
When you told me there was a setting I did not know existed, I thought, 'how did I miss this? I read the docs. I downloaded the example config, and carefully checked every entry.'
So I search this repo for that string, and aisde from two entries in code, there is only one mention of this excellent feature in the docs - that list in the readme.

I know it can be annoying when users don't RTFM, but I did, and I still failed and was an annoying user, so I'm trying to make sure that doesn't happen again. I hope this is helpful!
